shell/common/shell_test_platform_view.cc

```diff
@@ -0,0 +1,32 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/common/shell_test_platform_view.h"
+
+#if OS_FUCHSIA
+#include "flutter/shell/common/shell_test_platform_view_vulkan.h"
+#else
+#include "flutter/shell/common/shell_test_platform_view_gl.h"
+#endif
+
+namespace flutter {
+namespace testing {
+
+std::unique_ptr<ShellTestPlatformView> ShellTestPlatformView::Create(
```

Comment:
We should really aim for all our tests to be as cross platform as possible. Towards this end, there is no reason that Vulkan can only be tested on Fuchsia (since it is either available or can be emulated everywhere). I suggest making this configurable via an enum that the test can select at runtime (or skip if not available). I was undertaking just such an effort that would collide with this one so I have closed it. But feel free to refer to https://github.com/flutter/engine/pull/14023 for a guide on my thought process.

Reply:
So @iskakaushik suggested I remove the Vulkan stubs entirely for now, land the changes to the rest of the ShellTestPlatformView, and iterate on the rendering API runtime changes in a future PR.

My plan was to have Vulkan not tied to Fuchsia and this was more of a stepping stone towards getting there, but after Kaushik's suggestion I agree it'd be better to land this in distinct pieces so we don't end up with behemoth patches.